### ISSUE TITLE
NumberInput (with unit) value overlaps label

### ISSUE THREAD

**Issue by jpveooys** [Royal-Navy/design-system#1546]
### Steps to reproduce

1. Render (on master):

    ```jsx
   <NumberInput
     name="number-input"
     onChange={() => {}}
     label="Label"
     value={1000}
     unit="unit"
   />
    ```

This is a regression, possibly in https://github.com/Royal-Navy/design-system/pull/1511

#### master

![image](https://user-images.githubusercontent.com/66470099/95839234-ad330b80-0d3a-11eb-88c8-474e7b462bb5.png)

#### 2.15.0

![image](https://user-images.githubusercontent.com/66470099/95839294-c2a83580-0d3a-11eb-999f-43b27fb4dde5.png)

####

**Comment by jpveooys** [comment]
Also note the case when the value is deleted:

![image](https://user-images.githubusercontent.com/66470099/95845969-bf18ac80-0d42-11eb-8e9f-a0d8fddba55f.png)

**Comment by thyhjwb6** [comment]
There was [a PR](https://github.com/Royal-Navy/design-system/pull/1511) recently. Anticipating that these issues are fixed in `2.16.0`.

**Comment by thyhjwb6** [comment]
This has been reproduced and it is an issue when there is a `unit` and a `label`.

<img width="114" alt="Screenshot 2020-10-14 at 09 51 32" src="https://user-images.githubusercontent.com/56078793/95966278-d8812d80-0e02-11eb-895c-5eff075261c0.png">
